back/engines/commercial/analytics/spec/jobs/analytics/import_latest_matomo_data_job_spec.rb

```diff
@@ -69,12 +69,59 @@
     end
   end
 
-  it 'raises an error if the configured matomo site is the default one' do
-    stub_const('ENV', ENV.to_h.merge(
-      'DEFAULT_MATOMO_TENANT_SITE_ID' => AppConfiguration.instance.settings('matomo', 'tenant_site_id')
-    ))
+  context 'when tenant has active lifecycle' do
+    it 'raises an error if the configured matomo site is the default one' do
+      stub_const('ENV', ENV.to_h.merge(
+        'DEFAULT_MATOMO_TENANT_SITE_ID' => AppConfiguration.instance.settings('matomo', 'tenant_site_id')
+      ))
+
+      expect(AppConfiguration.instance.settings.dig('core', 'lifecycle_stage')).to eq('active')
+      expect { described_class.perform_now(Tenant.current.id) }
+        .to raise_error(described_class::MatomoMisconfigurationError)
+    end
+  end
+
+  context 'when tenant has trial lifecycle' do
+    before do
+      AppConfiguration.instance.settings['core']['lifecycle_stage'] = 'trial'
+      AppConfiguration.instance.save!
+    end
+
+    after do
+      AppConfiguration.instance.settings['core']['lifecycle_stage'] = 'active'
+      AppConfiguration.instance.save!
+    end
+
+    it 'raises an error if the configured matomo site is the default one' do
+      stub_const('ENV', ENV.to_h.merge(
+        'DEFAULT_MATOMO_TENANT_SITE_ID' => AppConfiguration.instance.settings('matomo', 'tenant_site_id')
+      ))
+
+      expect(AppConfiguration.instance.settings.dig('core', 'lifecycle_stage')).to eq('trial')
+      expect { described_class.perform_now(Tenant.current.id) }
+        .to raise_error(described_class::MatomoMisconfigurationError)
+    end
+  end
+
+  context 'when tenant has demo lifecycle' do
+    before do
+      AppConfiguration.instance.settings['core']['lifecycle_stage'] = 'demo'
+      AppConfiguration.instance.save!(validate: false)
+    end
 
-    expect { described_class.perform_now(Tenant.current.id) }
-      .to raise_error(described_class::MatomoMisconfigurationError)
+    after do
+      AppConfiguration.instance.settings['core']['lifecycle_stage'] = 'active'
+      AppConfiguration.instance.save!(validate: false)
+    end
+
+    it 'does not raise an error if the configured matomo site is the default one' do
+      stub_const('ENV', ENV.to_h.merge(
+        'DEFAULT_MATOMO_TENANT_SITE_ID' => AppConfiguration.instance.settings('matomo', 'tenant_site_id')
+      ))
+
+      expect(AppConfiguration.instance.settings.dig('core', 'lifecycle_stage')).to eq('demo')
+      expect { described_class.perform_now(Tenant.current.id) }
+        .not_to raise_error(described_class::MatomoMisconfigurationError)
```

Comment:
```suggestion
      expect { described_class.perform_now(Tenant.current.id) }.not_to raise_error
```

Reply:
I instead specified the error that occurs following the execution of the code being tested (Matomo::Client::MissingBaseUriError) + added an explanatory comment